.editorconfig

```diff
@@ -1,27 +1,94 @@
-# Summary: coding style configuration for editors that read .editorconfig.
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Common editor configurations for this project.
 #
 # EditorConfig defines a file format for specifying some common coding style
-# parameters. Many editors recognize .editorconfig files automatically, and
-# there exist plugins for other editors. See https://spec.editorconfig.org/.
+# parameters. Many IDEs and editors read .editorconfig files, either natively
+# or via plugins. A few formatters also read .editorconfig; shfmt and Prettier
+# are two examples (as of early 2025).
+#
+# We mostly follow Google's style guides (https://google.github.io/styleguide/)
+# with very few deviations.
+#
+# Miscellaneous notes:
+#
+# - The EditorConfig property `max_line_length` is not set here because its
+#   intended behavior is poorly specified. (See the discussion in the comments
+#   at https://github.com/editorconfig/editorconfig/issues/387) It *would* have
+#   been desirable to define a project convention for the line width here, but
+#   we must instead use editor-specific configuration files to do that.
+#
+# - With few exceptions (e.g., shfmt), `.editorconfig` files are not read by
+#   linters or formatters, which means the project needs separate config files
+#   for those tools. This includes markdownlint, yamllint, and others.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 root = true
 
 [*]
 charset = utf-8
-end_of_line = lf
+indent_style = space
 insert_final_newline = true
 spelling_language = en-US
+# It would be preferable not to set tab_width, but some EditorConfig plugins
+# (e.g., Emacs's) set it equal to indent_size if it's not set otherwise.
+tab_width = 8
```

Comment:
Please delete, it is the common default.  Moreover with `indent_style` set to space the  tab characters should not appear in the source.  If Emacs does something funny, it should be up to its plugin or user config to fix that.